packages/core/src/storage/vectorStore/utils.ts

```diff
@@ -16,20 +17,17 @@ export function nodeToMetadata(
   textField: string = DEFAULT_TEXT_KEY,
   flatMetadata: boolean = false,
 ): Metadata {
-  const nodeObj = node.toJSON();
-  const metadata = node.metadata;
+  const { metadata, embedding, ...rest } = _.cloneDeep(node.toJSON());
```

Comment:
@marcusschiesser we should remove lodash and use structuredClone in the future, but right now we need to support Node 16 (unofficially)